spi/transfer-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/transfer/TransferProcess.java

```diff
@@ -136,12 +138,19 @@ public void addProvisionedResource(ProvisionedResource resource) {
     }
 
     public boolean provisioningComplete() {
-        if (resourceManifest == null || provisionedResourceSet == null || resourceManifest.empty() || provisionedResourceSet.empty()) {
+        if (resourceManifest == null) {
```

Comment:
Personally, I prefer having explicit `if-`checks, because they avoid potentially unneeded Stream operations: if the `provisionedResourceSet` is null or empty, the `definitions` variable does not need to get created via stream ops. The overall behaviour stays the same though, so feel free to disregard :)

Reply:
fine for me the way it is.